---
title: "DTUResultsReport"
author: "Kristen Steenbergen"
date: "28/04/2021"
output:
  html_document:
      theme: spacelab
      code_folding: hide
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, message = FALSE, warning = FALSE)

library(tximport)
library(IsoformSwitchAnalyzeR)
```

# Introduction

Introductory paragraph here.  So far this is a summary of the IsoformSwitchAnalyzeR DTU Analysis Workflow.

# IsoformSwitchAnalyzeR Analysis:
## Step 1 - Import Data

The Salmon quantification data was imported to R using tximport, and the

```{r Import}
#Step 1 - Import Data:

# Import data from Salmon Quantification
dir <-  "../../data/rnaseq_TC32_MSC_E7107/salmon_out/"
salmonQuant <- importIsoformExpression(parentDir = dir, ignoreAfterPeriod = TRUE)
head(salmonQuant$counts)

#Create a data frame with sample names, and condition - Removed cell column that was use in DGE
sampleTable <- data.frame(
  sampleID = colnames(salmonQuant$counts[2:13]),
  condition = factor(rep(c("NT", "E7107"), times = 2, each = 3)))
sampleTable

# Create switchAnalyzeRlist
### for isoformexonannoation argument - use the 97th release, newest release of GTF files isn't successful (Jaccard similarity is too low)
theSwitchList <- importRdata(
  isoformCountMatrix   = salmonQuant$counts,
  isoformRepExpression = salmonQuant$abundance,
  designMatrix         = sampleTable,
  isoformExonAnnoation = "Homo_sapiens.GRCh38.97.chr_patch_hapl_scaff.gtf.gz",
  #isoformNtFasta = '',
  showProgress = TRUE,
  ignoreAfterPeriod = TRUE,
  removeNonConvensionalChr = TRUE)
```

## Step2 - Filtering the Data and Identifying Differentially used Isoforms

Following the steps through the outlined workflow, the data is filtered out. The geneExpressionCutoff argument defaults to 1, but this analysis was run with the threshold of 5 TPM.  The filter removed 28% (37,000) of the transcripts  and 92,000 remained in the analysis.  The higher threshold was chosen because if the cutoff is too low, there is a higher false discovery rate regarding the isoform fraction (IF) which is an important measure in the DTU analysis.

The differentially used isoforms were identified using a SwitchTest run with DEXSeq using default arguments for all except "reducetoSwitchingGenes" was set to FALSE in preparation for downstream global alternative splicing analysis.

``` {r filtering and Analyzing}
#These are defaults (except gene expr) - need to adjust?
#Set geneExpressionCutoff to 5 TPM, see details regarding Isoform Fraction (IF)
#Removed 37K(28%) transcripts with 92K isoforms left
filteredSwitchList <- preFilter(
  switchAnalyzeRlist = theSwitchList,
  geneExpressionCutoff = 5,
  isoformExpressionCutoff = 0,
  IFcutoff = 0,
  removeSingleIsoformGenes = TRUE)

# Step 3 - Identifying differentially used isoforms (SwitchTest)
IsoSwSwitchList <- isoformSwitchTestDEXSeq(
  switchAnalyzeRlist = filteredSwitchList,
  alpha = .05,
  dIFcutoff = .01,
  reduceToSwitchingGenes = FALSE)
extractSwitchSummary(IsoSwSwitchList)
```
The summary of the SwitchTest shows that there are 1882 isoforms identified, 819 switches with 1462 genes (These numbers may need to be updated as analysis is tweaked)

## Step 4 - Analyze Alternative Splicing

Alternate Splicing feature analysis was added to the SwitchList using default settings.  
The table below shows how many isoforms containing the given number of intron retentions.
```{r AltSplicing Analysis, message=FALSE}
#Alternative Splicing Function:
AnalyzedSwitchList <- analyzeAlternativeSplicing(
  switchAnalyzeRlist = IsoSwSwitchList,
  onlySwitchingGenes=FALSE,
  alpha=0.05,
  dIFcutoff = 0.1)
table(AnalyzedSwitchList$AlternativeSplicingAnalysis$IR)
#results: 3226 had 1 retained intron, 84 had 4 IR, one isoform had 12 retained introns

#Intron Retention Function:
IntronSwitchList <- analyzeIntronRetention(
  switchAnalyzeRlist = IsoSwSwitchList,
  onlySwitchingGenes = TRUE,
  alpha = 0.05,
  dIFcutoff = 0.1)
```

## Step 5 - Global Splicing Analysis

In this step the summaries of the above results are extracted using the default settings.  Because a global splicing analysis was of interest, the 'switchingGenes' arguments in the functions above were set to false.  

Include plot details here:
``` {r GW Alt Splicing}
#Genome-Wide Analysis of Alternative SPlicing - ust AnalyzedSwitchLIst for this
AnalyzedSwitchList
SpliceSummary <- extractSplicingSummary( AnalyzedSwitchList )
SpliceSummary
SplicingEnrichment <- extractSplicingEnrichment( AnalyzedSwitchList )
SplicingEnrichment
#extractSplicingEnrichmentComparison( AnalyzedSwitchList) - Only one condition so this cannot be completed
GWSummary <- extractSplicingGenomeWide( AnalyzedSwitchList,
                                        featureToExtract = 'all',
                                        splicingToAnalyze = c('A3', 'A5', 'IR', 'ES', 'ATSS', 'ATTS'),
                                        violinPlot = FALSE)
GWSummary
```
