## Kristen Steenbergen Portfolio

This website contains some of my work throughout my time in graduate school at Johns Hopkins University, 2019-2022.

This portfolio is in response to lack of ability to *show* skills rather than just tell about them.

This portfolio will be periodically updated
Areas of skill include R, Python, SQL, etc.

Projects listed are:

### h-CoV-19 Variant Mutation Browser
<www.covidvariantbrowser.com> is an application I developed to compare and display distinguishing SNP mutations of different variants of h-CoV-2 (SARS-CoV-2).   The mutations can be viewed as a side-by-side comparison of 2 variants, or of all the variant mutations present on each gene.  This project was completed as a final project for the *Advanced Practical Computer Concepts* class and utilized a wide array of web development elements such as HTML, CSS, Jquery, python CGI scripts, and SQL.  
  
**This tool was last updated May 1, 2021, and will be soon updated to include both the Delta and Omicron Variants**

  
### R-loops and Spilcing Gene Expression Research Project
<img align="right" src="images/Portfolio_Rsearch.png"> This volunteer research experience involved the analysis of gene expression datasets of Ewing sarcoma cells treated with a splicing inhibitor.  The transcriptomes were assessed to explore how a splicing deficiency in Ewing sarcoma impacts R-loops.  I completed the differential gene expression analysis along with the differential transcript usage analysis in R and prepared the following reports in RMarkdown to be presented at the lab meetings.  

<!--- This is HTML code that works in markdown when you want to do something specific - these reports open in a new tab, but it may be better to have all togehter for better user experience?  Then it can be executed with: [Differential Gene Expression (DGE) Report](./DGE_MarkdownReport.html) It's just that they take a while to load -->
Reports:  
<a href="./DGE_MarkdownReport.html" target="_blank">Differential Gene Expression (DGE) Report</a>   
<a href="./DTU_MarkdownReport.html" target="_blank">Differential Transcript Usage (DTU) Report</a>  

Skills devloped through this project include the further development of advanced R programming skills, usage of Bioconductor packages, utilization of NGS data, and the application of biostatisics.

This research project was facilitated by the Bioinformatics Research Network [(BRN)](https://www.bio-net.dev).  In order to to participate in a research project through the BRN,  proof of competency is required by completiing mini-projects.  The following RNA-Seq and ChIP-Seq analyses projects were completed and can be found here:  
<a href="https://rpubs.com/KSteenbergen/738407/" target="_self">RNA-Seq Analysis</a>   
<a href="https://rpubs.com/KSteenbergen/743415/" target="_self">ChIP-Seq Analysis</a>
  
### Manuscript Paper:    
The following manuscript was completed for a group project for the *Bioinformatics: Tools for Genome Analysis* class.  As a group we researched the heritability of a novel SNP (rs291014) in the miRNA146A gene that may be associated with an increasd risk of Behcet's disease.  This was a semester-long project that mimicked the process of peer-review journal publication while utilizing publicly available genomic data.  The final manuscript was formatted with LaTeX software using the PLOS template in Overleaf.  
*Please note, this paper has not be submitted for publication.*    
  
**Manuscript uploaded after final submission on December 20,2021**  
  
This project was an excellent opportunity to further develop technical writing skills, along with group communication and collaboration skills.  Additional technical skills include LaTeX, increased familiarity with various tools and file formats, NGS data processing consisting of quality control assessment, sequence alignment, and variation calling.  


### GFF3 Feature Exporter Program in Python
Since GFF3 is a common file format in bioinformatics, for one of the elements of the final exam for my *Practical Computer Concepts* class involved writing a program in python that would parse a GFF3 file for a given feature and then output the sequence in FASTA format.  This program executes with 4 arguments indicating the GFF3 file, type, ID, and value of the feature of interest. 
The attached video shows the FASTA output for the YAL068W-A gene, and then the coding sequence of the YAL068C gene in the *S.cerevisiae* genome.  Both of the genes are located on chromosome I. 

*Note: Because this program was written for an exam, for plagiarism reasons the code is not publicly shared.  However, I am happy to share the program code to anyone who is not a fellow JHU classmate in the bioinformatics/biotechnology program.*


